tests/test_upload_filename_normalization.py

```
@@ -0,0 +1,134 @@
+    "fastapi not installed (optional integration dependency)",
+)
+class UploadFilenameNormalizationTests(unittest.TestCase):
+    """Keep Windows and POSIX upload names canonical and workspace-contained."""
+
+    def test_canonical_basename_handles_both_separator_families_and_fallbacks(self):
+        """One helper normalizes client paths and rejects empty relative names."""
+        cases = {
+            r"..\..\etc\passwd": "passwd",
+            "../../etc/passwd": "passwd",
+            r"C:\recordings\meeting.wav": "meeting.wav",
+            "/tmp/recordings/meeting.wav": "meeting.wav",
```

Ruff (0.16.1)

[error] Probable insecure usage of temporary file or directory: "/tmp/recordings/meeting.wav"

(S108)